src/main/java/ch/uzh/ifi/hase/soprafs21/service/JwtTokenUtil.java

```diff
@@ -0,0 +1,54 @@
+package ch.uzh.ifi.hase.soprafs21.service;
+
+import ch.uzh.ifi.hase.soprafs21.security.config.SecurityConstants;
+import io.jsonwebtoken.*;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+
+@Service
+public class JwtTokenUtil {
```

Comment:
Shouldn't we delegate the token management to the Google identity provider? From the links I've shared, the examples show that we shouldn't do anything with the tokens, except maybe getting profile data from the access token.

Reply:
Yes, it can be done in that way. And custom token generation and validation are another way of doing the same thing. Additionally, we can also reuse this for adding another provider for authentication